language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/grpc/CommandsSubmission.java

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.rxjava.grpc;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static java.util.Optional.empty;
+
+import com.daml.ledger.javaapi.data.codegen.HasCommands;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * This class should be used to build a valid submission. It provides {@link #create(String, String, List)}
+ * for initial creation and extra builder methods for additional parameters
+ * e.g {@link #withActAs(List)}, {@link #withWorkflowId(String)} e.t.c
```

Comment:
```suggestion
 * e.g {@link #withActAs(List)}, {@link #withWorkflowId(String)} etc.
```
Not a native speaker so feel free to challenge me here but I've never seen `etc.` spelled out that way.